plugins/pulp_rpm/plugins/importers/yum/depsolve.py

```diff
@@ -250,11 +269,7 @@ def _build_source_with_provides(self):
         """
         fields = list(models.RPM.unit_key_fields)
         fields.extend(['provides', 'version_sort_index', 'release_sort_index'])
-        units = repo_controller.find_repo_content_units(
-            repository=self.source_repo,
-            repo_content_unit_q=mongoengine.Q(unit_type_id=ids.TYPE_ID_RPM),
-            unit_fields=fields, yield_content_unit=True
-        )
+        units = self._unit_generator(fields)
```

Comment:
As the comment in `_unit_generator` indicates, this got split out so that we could mock this return instead of having to mock repo_controller.